test/.gitattributes

```diff
@@ -0,0 +1,4 @@
+# Remove the text attribute from reference files, so that git doesn't convert
+# line separators on Windows machines. It causes the index files to become out
+# of sync with the fasta files.
+*.fa* -text
```

Comment:
Clearly by _regenerate_ I'm referring to having _test.pl_ generate the derived _.fai_ files (and removing them from source control), just as samtools's _test.pl_'s `test_index()` generates some _.bai_ indices used by later tests.  Additions to _test/test_view.c_ or a new _test/test_fai.c_ would indeed be needed to exercise `fai_build()`, but this would increase HTSlib's standalone test coverage so would obviously be a good thing…